plugins/module_utils/mso.py

```diff
@@ -294,96 +327,119 @@ def __init__(self, module):
+            self.params["output_level"] = "debug"
 
-        if self.module._socket_path is None:
-            if self.params.get('use_ssl') is None:
-                self.params['use_ssl'] = True
-            if self.params.get('use_proxy') is None:
-                self.params['use_proxy'] = True
-            if self.params.get('validate_certs') is None:
-                self.params['validate_certs'] = True
+        # Ensure protocol is set
+        self.params["protocol"] = "https" if self.params.get("use_ssl", True) else "http"
 
-            # Ensure protocol is set
-            self.params['protocol'] = 'https' if self.params.get('use_ssl', True) else 'http'
+        if self.params.get("host") is None:
+            self.fail_json(msg="Parameter 'host' is required when not using the HTTP API connection plugin")
 
-            # Set base_uri
-            if self.params.get('port') is not None:
-                self.base_only_uri = '{protocol}://{host}:{port}/'.format(**self.params)
-                self.baseuri = '{0}api/v1/'.format(self.base_only_uri)
-            else:
-                self.base_only_uri = '{protocol}://{host}/'.format(**self.params)
-                self.baseuri = '{0}api/v1/'.format(self.base_only_uri)
+        # Set base_only_uri
+        if self.params.get("port") is not None:
+            self.base_only_uri = "{protocol}://{host}:{port}/".format(**self.params)
+        else:
+            self.base_only_uri = "{protocol}://{host}/".format(**self.params)
 
-            if self.params.get('host') is None:
-                self.fail_json(msg="Parameter 'host' is required when not using the HTTP API connection plugin")
+        # Set MSO params
+        if self.module._socket_path is None:
+            if self.params.get("use_ssl") is None:
+                self.params["use_ssl"] = True
+            if self.params.get("use_proxy") is None:
+                self.params["use_proxy"] = True
+            if self.params.get("validate_certs") is None:
+                self.params["validate_certs"] = True
 
-            if self.params.get('password'):
-                # Perform password-based authentication, log on using password
-                self.login()
-            else:
-                self.fail_json(msg="Parameter 'password' is required for authentication")
+            # Set base_uri
+            self.baseuri = "{0}api/v1/".format(self.base_only_uri)
+        # Set NDO params
         else:
             self.connection = Connection(self.module._socket_path)
             if self.connection.get_platform() == "cisco.nd":
                 self.platform = "nd"
 
+            # Set base_uri
+            self.baseuri = "{0}mso/api/v1/".format(self.base_only_uri)
+
+        if self.params.get("password"):
+            # Perform password-based authentication, log on using password
+            self.login()
+        else:
+            self.fail_json(msg="Parameter 'password' is required for authentication")
```

Comment:
You can't move this outside of "if self.module._socket_path is None:" as this only applies to non-HTTPAPI plugin cases.